app/assets/javascripts/services/queriesSvc.js

```diff
@@ -51,23 +57,47 @@ angular.module('QuepidApp')
       svc.reset = reset;
       function reset() {
         svc.queries = {};
+        svc.showOnlyRated = false;
         svc.svcVersion++;
       }
 
       this.getCaseNo = getCaseNo;
       this.createSearcherFromSettings = createSearcherFromSettings;
+      this.normalizeDocExplains = normalizeDocExplains;
+      this.toggleShowOnlyRated = toggleShowOnlyRated;
 
       svc.bootstrapQueries = bootstrapQueries;
+      svc.showOnlyRated = false;
 
-      function createSearcherFromSettings(passedInSettings, query) {
+      function createSearcherFromSettings(passedInSettings, queryText, query) {
         var args = angular.copy(passedInSettings.selectedTry.args);
 
         if (passedInSettings && passedInSettings.selectedTry) {
+          // Modify query if ratings were passed in
+          if (query) {
+            if (passedInSettings.searchEngine === 'es') {
+              var mainQuery = args['query'];
+              args['query'] = {
+                'bool': {
+                  'should': mainQuery,
+                  'filter': query.filterToRatings(passedInSettings, true)
+                }
+              };
+            } else {
```

Comment:
should we explicitly check Solr?    and then have a else clause that throws an exception?   Going to soon have OpenSearch ;-)